Add Scale conftest.py

sarahbx opened the pull request:
Add `tests/scale/conftest.py`

Part of a dependency path for scale upgrade testing #3031
Rebased on #3105 

Depends on:
#3029 - Move Chaos utils
#3103 - Move EUS Constant
#3105 - Add scale utils.py

##### Short description:

##### More details:

##### What this PR does / why we need it:

##### Which issue(s) this PR fixes:

##### Special notes for reviewer:

##### jira-ticket:
<!--  full-ticket-url needs to be provided. This would add a link to the pull request to the jira and close it when the pull request is merged
If the task is not tracked by a Jira ticket, just write "NONE".
-->


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added scale testing infrastructure and fixtures for large-cluster scenarios.
  * Added HTTP-based VM service reachability helpers for test validation.
  * Added a test fixture to enforce a fork-based multiprocessing start method during tests.

* **Refactor**
  * Consolidated and reorganized test utilities into shared modules with clearer APIs.
  * Made MCP update waiting configurable with timeout/sleep options.

* **Chores**
  * Updated linter configuration and centralized constants (added EUS).

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

sarahbx committed:
Move chaos utils

Move chaos utils for use in other modules
- create_vm_with_nginx_service
- verify_vm_service_reachable
- is_http_ok

sarahbx committed:
Add typing, rename args

sarahbx committed:
Move EUS constant to fix import error

sarahbx committed:
Add scale utils.py

sarahbx committed:
Add scale/conftest.py

openshift-virtualization-qe-bot-3 commented:
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

## Welcome! 🎉

This pull request will be automatically processed with the following features:

### 🔄 Automatic Actions
* **Reviewer Assignment**: Reviewers are automatically assigned based on the OWNERS file in the repository root
* **Size Labeling**: PR size labels (XS, S, M, L, XL, XXL) are automatically applied based on changes
* **Issue Creation**: A tracking issue is created for this PR and will be closed when the PR is merged or closed
* **Pre-commit Checks**: [pre-commit](https://pre-commit.ci/) runs automatically if `.pre-commit-config.yaml` exists
* **Branch Labeling**: Branch-specific labels are applied to track the target branch
* **Auto-verification**: Auto-verified users have their PRs automatically marked as verified

### 📋 Available Commands

#### PR Status Management
* `/wip` - Mark PR as work in progress (adds WIP: prefix to title)
* `/wip cancel` - Remove work in progress status
* `/hold` - Block PR merging (approvers only)
* `/hold cancel` - Unblock PR merging
* `/verified` - Mark PR as verified
* `/verified cancel` - Remove verification status
* `/reprocess` - Trigger complete PR workflow reprocessing (useful if webhook failed or configuration changed)

#### Review & Approval
* `/lgtm` - Approve changes (looks good to me)
* `/approve` - Approve PR (approvers only)
* `/automerge` - Enable automatic merging when all requirements are met (maintainers and approvers only)
* `/assign-reviewers` - Assign reviewers based on OWNERS file
* `/assign-reviewer @username` - Assign specific reviewer
* `/check-can-merge` - Check if PR meets merge requirements

#### Testing & Validation
 * `/retest tox` - Run Python test suite with tox
 * `/retest build-container` - Rebuild and test container image
 * `/retest all` - Run all available tests


#### Container Operations
* `/build-and-push-container` - Build and push container image (tagged with PR number)
  * Supports additional build arguments: `/build-and-push-container --build-arg KEY=value`

#### Cherry-pick Operations
* `/cherry-pick <branch>` - Schedule cherry-pick to target branch when PR is merged
  * Multiple branches: `/cherry-pick branch1 branch2 branch3`

#### Label Management
* `/<label-name>` - Add a label to the PR
* `/<label-name> cancel` - Remove a label from the PR

### ✅ Merge Requirements

This PR will be automatically approved when the following conditions are met:

1. **Approval**: `/approve` from at least one approver
2. **LGTM Count**: Minimum 2 `/lgtm` from reviewers
3. **Status Checks**: All required status checks must pass
4. **No Blockers**: No WIP, hold, or conflict labels
5. **Verified**: PR must be marked as verified (if verification is enabled)

### 📊 Review Process

<details>
<summary><strong>Approvers and Reviewers</strong></summary>


**Approvers:**
 * dshchedr
 * myakove
 * rnetser
 * vsibirsk


**Reviewers:**
 * OhadRevah
 * RoniKishner
 * albarker-rh
 * dshchedr
 * geetikakay
 * guchen11
 * hmeir
 * qwang1
 * rlobillo
 * rnetser
 * sarahbx
 * vsibirsk


</details>

<details>
<summary><strong>Available Labels</strong></summary>

 * hold
 * verified
 * wip
 * lgtm
 * approve
 * automerge

</details>

### 💡 Tips

* **WIP Status**: Use `/wip` when your PR is not ready for review
* **Verification**: The verified label is automatically removed on each new commit
* **Cherry-picking**: Cherry-pick labels are processed when the PR is merged
* **Container Builds**: Container images are automatically tagged with the PR number
* **Permission Levels**: Some commands require approver permissions
* **Auto-verified Users**: Certain users have automatic verification and merge privileges

For more information, please refer to the project documentation or contact the maintainers.

sarahbx committed:
Add typing, make name required

sarahbx committed:
Alter chaos multiprocessing start method

The previous fix of using get_context(method) is no longer working.
To remedy this, switching to a fixture that sets the
multiprocessing start method instead.
https://github.com/python/cpython/issues/132898
https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Process

sarahbx committed:
Add docstring to get_user_kubeconfig_context

sarahbx committed:
Add MachineConfigPoolConfiguration ExitStack

sarahbx committed:
Refactor to use MachineConfigPoolConfiguration

sarahbx committed:
Update unittests for operator.py

sarahbx committed:
Add additional unittetst assertion

sarahbx commented on tests/scale/conftest.py:
```diff
@@ -0,0 +1,170 @@
+        wait_for_hco_conditions(
+            admin_client=admin_client,
+            hco_namespace=hco_namespace,
+        )
```
I already have `wait_for_reconcile_post_update=True` set, so not sure why I have this here.

openshift-virtualization-qe-bot-3 commented:
/retest all

_Auto-triggered: Files in this PR were modified by merged PR #3965._

<details>
<summary>Overlapping files</summary>

utilities/virt.py

</details>
